src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -1406,33 +1406,29 @@ BOOL AwtWindow::UpdateInsets(jobject insets)
     } else {
         m_insets.top = -1;
     }
+
     if (m_insets.left < 0 || m_insets.top < 0 ||
         m_insets.right < 0 || m_insets.bottom < 0)
     {
         /* This window hasn't been sized yet -- use system metrics. */
         jobject target = GetTarget(env);
         if (IsUndecorated() == FALSE) {
             /* Get outer frame sizes. */
-            LONG style = GetStyle();
-            if (style & WS_THICKFRAME) {
-                m_insets.left = m_insets.right =
-                    ::GetSystemMetrics(SM_CXSIZEFRAME) + extraPaddedBorderInsets;
-                m_insets.top = m_insets.bottom =
-                    ::GetSystemMetrics(SM_CYSIZEFRAME) + extraPaddedBorderInsets;
-            } else {
-                m_insets.left = m_insets.right =
-                    ::GetSystemMetrics(SM_CXDLGFRAME) + extraPaddedBorderInsets;
-                m_insets.top = m_insets.bottom =
-                    ::GetSystemMetrics(SM_CYDLGFRAME) + extraPaddedBorderInsets;
-            }
+            // System metrics are same for resizable & non-resizable frame.
+            // SM_CXFRAME is the width of the horizontal border in pixels.
+            // SM_CYFRAME is the height of the vertical border in pixels.
```

Comment:
I guess the meaning of `CX/CYFRAME` is quite obvious.

The first sentence about the metrics being the same is helpful.